File: Easy/1. Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  let obj = {};

  for (let i = 0; i < nums.length; i++) {
      let currentDiff = target - nums[i];
      if (obj[currentDiff] !== undefined) {
          return [obj[currentDiff], i]
      } else {
          obj[nums[i]] = i;
      }
  }
};

File: Easy/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        obj = {}
        i = 0
        while i < len(nums):
            diff = target - nums[i]
            if diff in obj:
                return [obj[diff], i]
            else:
                obj[nums[i]] = i
            i += 1